debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9JavaVMHelper.java

```diff
@@ -34,8 +35,26 @@
 import com.ibm.j9ddr.vm29.pointer.generated.J9JavaVMPointer;
 import com.ibm.j9ddr.vm29.pointer.generated.J9MethodPointer;
 import com.ibm.j9ddr.vm29.pointer.generated.J9VMSystemPropertyPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.J9BuildFlags;
+import com.ibm.j9ddr.vm29.pointer.helper.J9ObjectHelper;
+import static com.ibm.j9ddr.vm29.structure.J9Consts.*;
 
 public class J9JavaVMHelper {
+	public static boolean compressObjectReferences;
```

Comment:
This field should be `final` like the one it replaces, or access should be via a method that can throw `CorruptDataException`.

Reply:
Agreed about final (I meant to do this but forgot), but I see no reason to have it throw - if we can't access the JavaVM in the static initializer, the world will stop there.

Reply:
No, jdmpview still has some use even if there's no JRE.
The user should continue to be able to do the same things as they could before without a JRE and we should be able to present the issue/failure in a user-friendly manner.

Reply:
This class deals enitrely with the J9JavaVM (every functions take a J9JavaVMPointer), so there's no way the clinit failure here could affect anything.

Reply:
Yes, but you've added this static field and numerous references to it, so the assumption that the client has a VM pointer is not safe.